### DISCUSSION TITLE
Custom Voice

### DISCUSSION THREAD

**Discussion by vsarda** [rand256/valetudo#361] [Q&A]
I see that you can upload new voice packs once rooted.  What's the easiest way to create a custom voice pkg file?  Or is it possible to upload a specific few wav files for the common sounds?

**Comment by thekarinka** [comment]
See https://github.com/dgiese/dustcloud/tree/master/devices/xiaomi.vacuum/original-soundpackages and https://github.com/dgiese/dustcloud/tree/master/devices/xiaomi.vacuum/audio_generator

**Reply by pktiuk** [reply]
@thekarinka   
https://github.com/dgiese/dustcloud/tree/master/devices/xiaomi.vacuum/original-soundpackages  redirects to http://dustbuilder.xvm.mit.edu/pkg/voice/ which does not exist.

EDIT I found it on https://builder.dontvacuum.me/pkg/voice/

**Comment by pidator** [comment]
Yes, you can accomplish this task either by
* simply replacing current wav files via ssh/scp in the `/opt/rockrobo/resources/sounds/` folder


or by
* creating a whole soundpack and upload it via the Valetudo WebGui. You may follow this little HowTo of @ccoors of the valetudo telegram channel for creating your own soundpack:

**Creating voice packs (Gen1/Gen2)**

You need a folder with .wav files (see below). Their content should correspond to the messages.

A wav-format known to work is "RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 44100 Hz"

From that folder create a tar archive:

```$ tar czf ../voice-pack.pkg .```

Encrypt that archive using ccrypt:

```$ ccrypt -e -K "r0ckrobo#23456" ../voice-pack.pkg```

You can then install the voice-pack.pkg as a normal voice pack in Valetudo (Settings -> Sound & Voice).

**List of required WAVs files** (tbc...)

- back_dock_failed.wav - "Docking station unreachable. Please remove obstacles and make sure the charging surfaces are clean!"
- back_dock_nearby.wav - "Docking station obstructed. Please remove obstacles!"
- bin_in.wav - "Dustbin installed"
- bin_out.wav - "Dustbin removed."
- binout_error10.wav - "Dustbin has been removed. Please clean the filter!"
- bl_recovery_bootfailed.wav - "Booting has failed, restoring initial setup. Restoration will take about 5 minutes. Please be patient."
- bl_recovery_failed.wav - "Restoration of initial setup failed. Please try again."
- bl_recovery_retry.wav - "Restarting initial setup. Startup will take about 5 minutes. Please be patient."
- bl_recovery_start.wav - "Restoring initial setup. Restoration will take about 5 minutes. Please be patient."
- bl_recovery_updatefailed.wav - "Upgrade failed. Restoration of initial setup will take about 5 minutes. Please be patient."
- charging.wav - Charging.
- clean_bin.wav - "Please clean the dustbin."
- clean_finish.wav - "Sanitary operations complete. Returning to dock."
- error_internal.wav - "An internal error occurred. Reboot the system."
- error1.wav - "Error #1: LIDAR unit jammed. Please rotate laser head to unjam."
- error2.wav - "Error #2: Bumper stuck. Please tap bumpers."
- error3.wav - "Error #3: Wheel suspended. Please move the device."
- error4.wav - "Error #4: Robot suspended. Please wipe the cliff sensor clean and move the device."
- error5.wav - "Error #5: Main brush jammed. Please remove and clean the main brush."
- error6.wav - "Error #6: Side brush jammed. Please remove and clean the side brushes."
- error7.wav - "Error #7: Main wheels jammed. Please remove obstructions and move the device."
- error8.wav - "Error #8: Device trapped. Please remove obstacles."
- error9.wav - "Error #9: Dustbin absent. Please install the dustbin and filter."
- error10.wav - "Error #10: Filter blocked. Please clean or replace the filter."
- error11.wav - "Error #11: High-intensity magnetic fields present. Please move the device."
- error12.wav - "Error #12: Battery low. Please charge the device."
- error13.wav - "Error #13: Can not charge. Please make sure the charging surfaces are clean."
- error14.wav - "Error #14: Battery malfunction."
- error15.wav - "Error #15: Wall sensor malfunction. Please wipe the wall sensor."
- error16.wav - "Error #16: Device is tilted. Please move the device to a flat surface."
- error17.wav - "Error #17: Side brush malfunction. Please reboot the device."
- error18.wav - "Error #18: Fan malfunction. Please reboot the device."
- error19.wav - "Error #19: Dock has no power. Please reconnect dock."
- error21.wav - "Error #21: Distance sensor malfunction. Please make sure that the top cover of the device is not pinned."
- error22.wav - "Error #22: Dock sensor failed. Please wipe contqact surfaces.."
- error23.wav - "Error #23: Dock signal lost. Please make sure the signal emission area of dock is clean."
- findme.wav - "Hello. Is anyone there? I can hear you but I can't see you."
- finish.wav - "Sanitation operation complete."
- goto.wav - "Target acquired. Initiating relocation procedure."
- goto_complete.wav - "Relocation procedure complete."
- goto_failed.wav - "Target lost. Terminating relocation procedure."
- home.wav - "Initiating docking operation."
- no_power.wav - "Battery level critical. Initiating docking operation."
- no_power_charging.wav - "Battery level critical. Do not interrupt charging operation."
- no_spot_on_dock.wav - "Please relocate the device prior to initiating local sanitary procedures."
- pause.wav - Paused.
- power_off.wav - "Shutting down. No hard feelings."
- power_off_rejected.wav - "Please remove the device from the dock prior to shut down."
- power_resume_clean.wav - "Battery fully operational. Resuming sanitary operations."
- relocate_failed.wav - "Do you know where you put me? Because I don't know where you put me."
- remote.wav - "Remote control initiated."
- remote_complete.wav - "Remote control terminated."
- restart_clean.wav - "Navigational lock lost. Restarting sanitary operations."
- resume_clean.wav - "Resuming sanitary operations."
- resume_home.wav - Charging.
- resume_zone.wav - "Resuming zone-specific sanitary operation."
- return_no.wav - "Initial position lost."
- return_yes.wav - "Target lost. Device is not close enough to the dock."
- spot.wav - "Initiating local sanitation procedures."
- start.wav - "Initiating sanitation procedures."
- stop_goto.wav - "Terminating relocation procedure."
- stop_spot.wav - "Terminating local sanitation procedures."
- stop_zone.wav - "Terminating zone-specific sanitation procedures."
- sysupd_complete.wav - "System updated."
- sysupd_failed.wav - "System update failed."
- sysupd_notready.wav - "Battery level critical. System update terminated."
- sysupd_start.wav - "Initiating system update."
- sysupd_wip.wav - "System update in progress."
- timed_clean.wav - "Initiating scheduled sanitary operations."
- wifi_reset.wav - "Resetting network connection."
- zone.wav - "Initiating zone-specific sanitary operations."
- zone_complete.wav - "Zoned sanitary operation complete. Initiating docking procedures."
- zone_failed.wav - "Zoned sanitary operation terminated. Initiating docking procedures."

**Reply by vsarda** [reply]
Thanks!  I have connected as root via SSH (I'm not sure how to connect as username ruby).  I don't see the sounds directory with the wav files.  Should it be at /mnt/data/rockrobo/?  Is it because I'm not logged in as ruby, or am I looking in the wrong place?

**Reply by pidator** [reply]
If you're using a newer prebuild image there is no user ruby. Nevertheless I've edited my first comment and added the missing path of the soundfiles:
`/opt/rockrobo/resources/sounds`

**Reply by rand256** [reply]
Installed via miio command sounds are placed at `/mnt/data/rockrobo/sounds` and are checked first. I guess you can just put your wav files there manually via SCP (creating the directory if it doesn't exist). If the requested sound is found there, device will play it, if it's not - device will play a sound from `/opt/rockrobo/resources/sounds/<loc>/`.

**Reply by xav4nte** [reply]
Great list @pidator 👍 but what about the “Positioning, please wait” phrase, do you know how to replace that?

**Reply by pidator** [reply]
> what about the “Positioning, please wait” phrase 

I assume it's the file `positioning.wav` in the named folder. That's why I stated a **tbc** behind the list topic 😃

**Reply by xav4nte** [reply]
Sorry, missed that part...

**Reply by sverdlyuk** [reply]
> Yes, you can accomplish this task either by
> 
> * simply replacing current wav files via ssh/scp in the `/opt/rockrobo/resources/sounds/` folder
> 
> or by
> 
> * creating a whole soundpack and upload it via the Valetudo WebGui. You may follow this little HowTo of @ccoors of the valetudo telegram channel for creating your own soundpack:
> 
> **Creating voice packs (Gen1/Gen2)**
> 
> You need a folder with .wav files (see below). Their content should correspond to the messages.
> 
> A wav-format known to work is "RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 44100 Hz"
> 
> From that folder create a tar archive:
> 
> `$ tar czf ../voice-pack.pkg .`
> 
> Encrypt that archive using ccrypt:
> 
> `$ ccrypt -e -K "r0ckrobo#23456" ../voice-pack.pkg`
> 
> You can then install the voice-pack.pkg as a normal voice pack in Valetudo (Settings -> Sound & Voice).
> 
> **List of required WAVs files** (tbc...)
> 
> * back_dock_failed.wav - "Docking station unreachable. Please remove obstacles and make sure the charging surfaces are clean!"
> * back_dock_nearby.wav - "Docking station obstructed. Please remove obstacles!"
> * bin_in.wav - "Dustbin installed"
> * bin_out.wav - "Dustbin removed."
> * binout_error10.wav - "Dustbin has been removed. Please clean the filter!"
> * bl_recovery_bootfailed.wav - "Booting has failed, restoring initial setup. Restoration will take about 5 minutes. Please be patient."
> * bl_recovery_failed.wav - "Restoration of initial setup failed. Please try again."
> * bl_recovery_retry.wav - "Restarting initial setup. Startup will take about 5 minutes. Please be patient."
> * bl_recovery_start.wav - "Restoring initial setup. Restoration will take about 5 minutes. Please be patient."
> * bl_recovery_updatefailed.wav - "Upgrade failed. Restoration of initial setup will take about 5 minutes. Please be patient."
> * charging.wav - Charging.
> * clean_bin.wav - "Please clean the dustbin."
> * clean_finish.wav - "Sanitary operations complete. Returning to dock."
> * error_internal.wav - "An internal error occurred. Reboot the system."
> * error1.wav - "Error [How to set up a no go Area or a fence #1](https://github.com/rand256/valetudo/issues/1): LIDAR unit jammed. Please rotate laser head to unjam."
> * error2.wav - "Error [SSH Key refused :/ #2](https://github.com/rand256/valetudo/issues/2): Bumper stuck. Please tap bumpers."
> * error3.wav - "Error [Zoned Cleaning #3](https://github.com/rand256/valetudo/issues/3): Wheel suspended. Please move the device."
> * error4.wav - "Error [Timed Zone Cleaning #4](https://github.com/rand256/valetudo/issues/4): Robot suspended. Please wipe the cliff sensor clean and move the device."
> * error5.wav - "Error [Customization User Settings Page? #5](https://github.com/rand256/valetudo/issues/5): Main brush jammed. Please remove and clean the main brush."
> * error6.wav - "Error [Can't get out of Cleaning History #6](https://github.com/rand256/valetudo/issues/6): Side brush jammed. Please remove and clean the side brushes."
> * error7.wav - "Error [Bump mqtt-packet from 5.6.0 to 5.6.1 #7](https://github.com/rand256/valetudo/pull/7): Main wheels jammed. Please remove obstructions and move the device."
> * error8.wav - "Error [Changing Zones Order may not function as intended #8](https://github.com/rand256/valetudo/issues/8): Device trapped. Please remove obstacles."
> * error9.wav - "Error [version GLIBCXX_3.4.21 not defined in file libstdc++.so.6 on RE4 #9](https://github.com/rand256/valetudo/issues/9): Dustbin absent. Please install the dustbin and filter."
> * error10.wav - "Error [update the MQTT client with valetudo remote branche #10](https://github.com/rand256/valetudo/pull/10): Filter blocked. Please clean or replace the filter."
> * error11.wav - "Error [Valetudo stopped working after latest Update RE4? #11](https://github.com/rand256/valetudo/issues/11): High-intensity magnetic fields present. Please move the device."
> * error12.wav - "Error [Zone Timers #12](https://github.com/rand256/valetudo/issues/12): Battery low. Please charge the device."
> * error13.wav - "Error [Verticel no go lines altered in the wrong direction #13](https://github.com/rand256/valetudo/issues/13): Can not charge. Please make sure the charging surfaces are clean."
> * error14.wav - "Error [Where to get FW 1886? #14](https://github.com/rand256/valetudo/issues/14): Battery malfunction."
> * error15.wav - "Error [Error: Unable to reach vacuum, handshake failed #15](https://github.com/rand256/valetudo/issues/15): Wall sensor malfunction. Please wipe the wall sensor."
> * error16.wav - "Error [Feature request: multiple cleaning passes for whole cleaning #16](https://github.com/rand256/valetudo/issues/16): Device is tilted. Please move the device to a flat surface."
> * error17.wav - "Error [No No-Go Zones would be saved? #17](https://github.com/rand256/valetudo/issues/17): Side brush malfunction. Please reboot the device."
> * error18.wav - "Error [Use the correct URL #18](https://github.com/rand256/valetudo/pull/18): Fan malfunction. Please reboot the device."
> * error19.wav - "Error [Update Dockerfile #19](https://github.com/rand256/valetudo/pull/19): Dock has no power. Please reconnect dock."
> * error21.wav - "Error [Feature request: option to disable the step-sensors #21](https://github.com/rand256/valetudo/issues/21): Distance sensor malfunction. Please make sure that the top cover of the device is not pinned."
> * error22.wav - "Error [Unknown error code #22](https://github.com/rand256/valetudo/issues/22): Dock sensor failed. Please wipe contqact surfaces.."
> * error23.wav - "Error [MQTT issue #23](https://github.com/rand256/valetudo/issues/23): Dock signal lost. Please make sure the signal emission area of dock is clean."
> * findme.wav - "Hello. Is anyone there? I can hear you but I can't see you."
> * finish.wav - "Sanitation operation complete."
> * goto.wav - "Target acquired. Initiating relocation procedure."
> * goto_complete.wav - "Relocation procedure complete."
> * goto_failed.wav - "Target lost. Terminating relocation procedure."
> * home.wav - "Initiating docking operation."
> * no_power.wav - "Battery level critical. Initiating docking operation."
> * no_power_charging.wav - "Battery level critical. Do not interrupt charging operation."
> * no_spot_on_dock.wav - "Please relocate the device prior to initiating local sanitary procedures."
> * pause.wav - Paused.
> * power_off.wav - "Shutting down. No hard feelings."
> * power_off_rejected.wav - "Please remove the device from the dock prior to shut down."
> * power_resume_clean.wav - "Battery fully operational. Resuming sanitary operations."
> * relocate_failed.wav - "Do you know where you put me? Because I don't know where you put me."
> * remote.wav - "Remote control initiated."
> * remote_complete.wav - "Remote control terminated."
> * restart_clean.wav - "Navigational lock lost. Restarting sanitary operations."
> * resume_clean.wav - "Resuming sanitary operations."
> * resume_home.wav - Charging.
> * resume_zone.wav - "Resuming zone-specific sanitary operation."
> * return_no.wav - "Initial position lost."
> * return_yes.wav - "Target lost. Device is not close enough to the dock."
> * spot.wav - "Initiating local sanitation procedures."
> * start.wav - "Initiating sanitation procedures."
> * stop_goto.wav - "Terminating relocation procedure."
> * stop_spot.wav - "Terminating local sanitation procedures."
> * stop_zone.wav - "Terminating zone-specific sanitation procedures."
> * sysupd_complete.wav - "System updated."
> * sysupd_failed.wav - "System update failed."
> * sysupd_notready.wav - "Battery level critical. System update terminated."
> * sysupd_start.wav - "Initiating system update."
> * sysupd_wip.wav - "System update in progress."
> * timed_clean.wav - "Initiating scheduled sanitary operations."
> * wifi_reset.wav - "Resetting network connection."
> * zone.wav - "Initiating zone-specific sanitary operations."
> * zone_complete.wav - "Zoned sanitary operation complete. Initiating docking procedures."
> * zone_failed.wav - "Zoned sanitary operation terminated. Initiating docking procedures."

Can you explain what the commands sound.info and sound.ver mean? What should be inside these sound files?

**Reply by pidator** [reply]
> Can you explain what the commands sound.info and sound.ver mean? What should be inside these sound files?

These two files aren't "sound files", I've just checked the two on my robot and the content of both is just a single number. But I don't know any additional information/meaning about these.

**Reply by sverdlyuk** [reply]
> > Can you explain what the commands sound.info and sound.ver mean? What should be inside these sound files?
> 
> These two files aren't "sound files", I've just checked the two on my robot and the content of both is just a single number. But I don't know any additional information/meaning about these.

Could you tell me which file contains the startup melody for the vacuum cleaner? Happy New Year !

**Reply by pidator** [reply]
> Could you tell me which file contains the startup melody for the vacuum cleaner? Happy New Year !

`/opt/rockrobo/resources/sounds/power_on.wav`

**Reply by sverdlyuk** [reply]
As far as I understand, the shutdown sound should be the file power_off.wav?

**Reply by pidator** [reply]
Just download the files using e.g. scp and listen to them on your workstation... wav files can be played by every media player.

**Reply by sverdlyuk** [reply]
Where can I download these files?

I am creating a custom voice pack for the Roborock vacuum cleaner. I downloaded an example of such a pack from here: https://builder.dontvacuum.me/pkg/voice/roborock/, but I am surprised that there is no voice file for turning on and off inside the voice pack. Are these files stored somewhere else, not in the voice pack?

**Reply by smartm0use** [reply]
I'm new here and I have an older robot (2020/2021). I've never heard my Xiaomi Gen1 say anything when turning on or off. Did you maybe mean the prompts for starting and stopping cleaning instead? If not, that might be why you couldn't find those voice files
